debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisMemoryThreshold.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server.redis;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.storage.redis.RedisClient;
+import io.debezium.util.IoUtil;
+import io.smallrye.mutiny.tuples.Tuple2;
+
+public class RedisMemoryThreshold implements Supplier<Boolean> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RedisMemoryThreshold.class);
+
+    private static final String INFO_MEMORY = "memory";
+    private static final String INFO_MEMORY_SECTION_MAXMEMORY = "maxmemory";
+    private static final String INFO_MEMORY_SECTION_USEDMEMORY = "used_memory";
+
+    private static final Supplier<Boolean> MEMORY_OK = () -> true;
+
+    private RedisClient client;
+
+    private int memoryThreshold;
+
+    private long memoryLimit;
+
+    private Supplier<Boolean> isMemoryOk;
+
+    public RedisMemoryThreshold(RedisClient client, int memoryThreshold, int memoryLimitMb) {
+        this.client = client;
+        this.memoryThreshold = memoryThreshold;
+        this.memoryLimit = 1024L * 1024 * memoryLimitMb;
+        Tuple2<Long, Long> memoryTuple;
+        if (memoryThreshold == 0 || (memoryTuple = memoryTuple(memoryLimit)) == null || memoryTuple.getItem2() == -1) {
```

Comment:
@ggaborg Please unwind this condition. I am not a great fan of assignments in ifs as they tend to be error-prone in long term run from maintenance PoV.